src/app/api/services/dot-menu.service.ts

```diff
@@ -78,10 +78,21 @@ export class DotMenuService {
      * @param portletId MenuItems id
      */
     getDotMenuId(portletId: string): Observable<string> {
+        return this.getDotMenu(portletId)
+            .map((menu: DotMenu) => menu.id);
+    }
+
+    getDotMenu(portletId: string): Observable<DotMenu> {
         return this.loadMenu()
             .flatMap((menus: DotMenu[]) => menus)
-            .find((menu: DotMenu) => menu.menuItems.some((menuItem) => menuItem.id === portletId))
-            .map((menu: DotMenu) => menu.id);
+            .find((menu: DotMenu) => menu.menuItems.some((menuItem: DotMenuItem) => menuItem.id === portletId));
+    }
+
+    getDotMenuItem(portletId: string): Observable<DotMenuItem> {
+        return this.loadMenu()
+            .flatMap((menus: DotMenu[]) => menus)
```

Comment:
Can't you just do `.flatMap((menus: DotMenu[]) => menus.menuItem)` and have only one `flatMap`?